gfx/video_crt_switch.c

```diff
@@ -0,0 +1,218 @@
+			aspect_ratio_switch(width,height);
+		}
+		if (height > 200 && height < 224){				/// for mame
+			aspect_ratio_switch(width,height);
+			ben_core_height = 224;
+		}
+		if (height > 224 && height < 240){				/// for mame
+			aspect_ratio_switch(width,height);
+			ben_core_height = 240;
+		}
+	
+		if (height > 240 && height < 255 ){								/// for mame
+			aspect_ratio_switch(width,height);
+			ben_core_height = 254;
+		}
+		if (height == 528 && set_ben_core_hz == 60){								/// for mame
+			aspect_ratio_switch(width,height);
+			ben_core_height = 480;
+		}
+	
+		if (height >= 240 && height < 255 && set_ben_core_hz == 55){
+			aspect_ratio_switch(width,height);
+			ben_core_height = 254;
+		}
+		
+	}
+	switch_res_crt(width, height);
+}
+
+void switch_res_crt(int width, int height){
+		
+	if ( height > 100){
+
+			switch_res(width, height,0);
+			ben_poke_video();
+			ben_tmp_height = ben_core_height;
+			ben_tmp_width = ben_core_width;
+	
+	}
+	
+}
+
+void aspect_ratio_switch(int width,int height){  // send aspect float to videeo_driver
+		fly_aspect = (float)width/height;
+		video_driver_set_aspect_ratio_value((float)fly_aspect);
+}
+
+void switch_crt_hz(){  // set hz float an int for windows switching
+	
+	if (ben_core_hz != ben_tmp_core_hz){
+		if (ben_core_hz < 53 ){
+			set_ben_core_hz = 50;	
+			
+		}
+		if (ben_core_hz >= 53  &&  ben_core_hz < 57){	
+			set_ben_core_hz = 55;	
+		}	
+		if (ben_core_hz >= 57 ){	
+			set_ben_core_hz = 60;	
+		}	
+		video_monitor_set_refresh_rate(ben_core_hz);
+		ben_tmp_core_hz = ben_core_hz;
+	}
+	
+}
+
+void switch_res(int width, int height, int f_restore){  // windows function to swith resolutions
+	
+	DEVMODE curDevmode;
+	DEVMODE devmode;
+	DWORD flags = 0;
+	int     iModeNum;
+	int depth = 0;
+	int freq;
+	if (f_restore == 0){
+		freq = set_ben_core_hz;
+	} else {
+		freq = 0;
+	}
+	
+	EnumDisplaySettings(NULL, ENUM_CURRENT_SETTINGS, &curDevmode);
+	
+	if (width == curDevmode.dmPelsWidth){
+		width  = 0;							// used to stop superresolution bug
+	}
+
+	if (width == 0) {
+		width = curDevmode.dmPelsWidth;
+	}
+	if (height == 0) {
+		height = curDevmode.dmPelsHeight;
+	}
+	if (depth == 0) {
+		depth = curDevmode.dmBitsPerPel;
+	}
+	if (freq == 0) {
+		freq = curDevmode.dmDisplayFrequency;
+	}
+
+	for (iModeNum = 0; ; iModeNum++) {
+		if (EnumDisplaySettings(NULL, iModeNum, &devmode)) {
+			// See if this setting has the right width.
+			if (devmode.dmPelsWidth != width) {
+				continue;
+			}
+
+			if (devmode.dmPelsHeight != height) {
+				continue;
+			}
+
+			// See if this setting has the right depth.
+			if (devmode.dmBitsPerPel != depth) {
+				continue;
+			}
+
+			// See if this setting has the right frequency.
+			if (devmode.dmDisplayFrequency != freq) {
+				continue;
+			}
+
+			// This new setting satisfies all the criteria. Try it.
+			devmode.dmFields |= DM_PELSWIDTH | DM_PELSHEIGHT | DM_BITSPERPEL | DM_DISPLAYFREQUENCY;
+			LONG res = ChangeDisplaySettings(&devmode, CDS_TEST);

```

Comment:
Variables not declared at the top of a function (before any other function calls or non-declarating assignments) breaks C89.